tests/v2/validation/snapshot/snapshot.go

```diff
@@ -4,148 +4,188 @@ import (
 	require.NoError(t, err)
 
-	err = workloads.VerifyDeployment(steveclient, deploymentResponse)
+	err = workloads.VerifyDeployment(steveclient, postDeploymentResp)
 	require.NoError(t, err)
-	require.Equal(t, WorkloadNamePostBackup, deploymentResponse.ObjectMeta.Name)
-	var restoreRKEConfig string
+	require.Equal(t, WorkloadNamePostBackup, postDeploymentResp.ObjectMeta.Name)
+
 	if upgrade {
-		clusterObject, clusterResponse, err := clusters.GetProvisioningClusterByName(adminClient, clusterName, namespace)
+		clusterObject, clusterResponse, err := clusters.GetProvisioningClusterByName(client, clusterName, namespace)
 		require.NoError(t, err)
 
-		var upgradeKubernetesVersion string
-		if strings.Contains(initialKubernetesVersion, "rke2") {
-			defaultVersion, err := kubernetesversions.Default(adminClient, clusters.RKE2ClusterType.String(), nil)
-			upgradeKubernetesVersion = defaultVersion[0]
-			require.NoError(t, err)
-		} else {
-			defaultVersion, err := kubernetesversions.Default(adminClient, clusters.K3SClusterType.String(), nil)
-			upgradeKubernetesVersion = defaultVersion[0]
-			require.NoError(t, err)
+		if upgradeKubernetesVersion == "" {
+			if strings.Contains(initialKubernetesVersion, RKE2) {
+				defaultVersion, err := kubernetesversions.Default(client, clusters.RKE2ClusterType.String(), nil)
+				upgradeKubernetesVersion = defaultVersion[0]
+				require.NoError(t, err)
+			} else {
+				defaultVersion, err := kubernetesversions.Default(client, clusters.K3SClusterType.String(), nil)
+				upgradeKubernetesVersion = defaultVersion[0]
+				require.NoError(t, err)
+			}
 		}
 
 		clusterObject.Spec.KubernetesVersion = upgradeKubernetesVersion
+
 		if strategy {
 			clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency = cpConcurrencyValue
 			clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency = workerConcurrencyValue
 		}
+
 		_, err = client.Steve.SteveType(ProvisioningSteveResouceType).Update(clusterResponse, clusterObject)
 		require.NoError(t, err)
+
 		err = clusters.WaitClusterToBeUpgraded(client, clusterID)
 		require.NoError(t, err)
 
 		podResults, podErrors = pods.StatusPods(client, clusterID)
 		assert.NotEmpty(t, podResults)
 		assert.Empty(t, podErrors)
-		require.NoError(t, err)
 		require.Equal(t, upgradeKubernetesVersion, clusterObject.Spec.KubernetesVersion)
 
 		if strategy {
 			require.Equal(t, cpConcurrencyValue, clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency)
 			require.Equal(t, workerConcurrencyValue, clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency)
 		}
-		restoreRKEConfig = "all"
 	}
+
 	snapshotRestore := &rkev1.ETCDSnapshotRestore{
 		Name:             snapshotToRestore,
 		Generation:       clusterObject.Spec.RKEConfig.ETCDSnapshotCreate.Generation,
-		RestoreRKEConfig: restoreRKEConfig,
+		RestoreRKEConfig: snapshotObj.SnapshotRestore,
```

Comment:
Why is this pulled from a config instead of from the cluster?

Reply:
This is being pulled from the new `snapshotInput` where a user can specify the type of snapshot restore that is going to happen. Before, there wasn't any flexibility

Reply:
Talked offline about this, leaving as is.